### ISSUE TITLE
[BUG] API errors shown for Huobi 

### ISSUE THREAD

**Issue by sweetpotatoz** [CoinAlpha/hummingbot#872]
**Describe the bug**
// A clear and concise description of what the bug is.

API error shown in the log.  But the bot managed to place trades once in a while.

in my XEMM bots
2019-10-04 06:17:33,002 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Limit bid order at 0.000677 btc is no longer profitable. Removing the order. [clock=2019-10-04 06:17:33+00:00]
2019-10-04 06:17:34,001 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Limit bid order at 0.000677 btc is no longer profitable. Removing the order. [clock=2019-10-04 06:17:34+00:00]
2019-10-04 06:17:34,001 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Cancelling the limit order buy-xzcbtc-1570166074002821. [clock=2019-10-04 06:17:34+00:00]
2019-10-04 06:17:34,077 - hummingbot.market.huobi.huobi_market - ERROR - Error received from https://api.huobi.pro/v1/order/orders/50580823106/submitcancel. Response is {'status': 'error', 'err-code': 'order-orderstate-error', 'err-msg': 'Incorrect order state', 'data': None, 'order-state': 5}.
2019-10-04 06:18:34,002 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Limit bid order at 0.000677 btc is no longer profitable. Removing the order. [clock=2019-10-04 06:18:34+00:00]
2019-10-04 06:18:35,003 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Limit bid order at 0.000677 btc is no longer profitable. Removing the order. [clock=2019-10-04 06:18:35+00:00]
2019-10-04 06:18:35,003 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Cancelling the limit order buy-xzcbtc-1570166074002821. [clock=2019-10-04 06:18:35+00:00]
2019-10-04 06:18:35,219 - hummingbot.market.huobi.huobi_market - ERROR - Error received from https://api.huobi.pro/v1/order/orders/50580823106/submitcancel. Response is {'status': 'error', 'err-code': 'order-orderstate-error', 'err-msg': 'Incorrect order state', 'data': None, 'order-state': 5}.
2019-10-04 06:19:35,004 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Limit bid order at 0.000677 btc is no longer profitable. Removing the order. [clock=2019-10-04 06:19:35+00:00]
2019-10-04 06:19:36,001 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Limit bid order at 0.000677 btc is no longer profitable. Removing the order. [clock=2019-10-04 06:19:36+00:00]
2019-10-04 06:19:36,001 - hummingbot.strategy.cross_exchange_market_making.cross_exchange_market_making - INFO - (xzcbtc) Cancelling the limit order buy-xzcbtc-1570166074002821. [clock=2019-10-04 06:19:36+00:00]
2019-10-04 06:19:36,077 - hummingbot.market.huobi.huobi_market - ERROR - Error received from https://api.huobi.pro/v1/order/orders/50580823106/submitcancel. Response is {'status': 'error', 'err-code': 'order-orderstate-error', 'err-msg': 'Incorrect order state', 'data': None, 'order-state': 5}.
2019-10-04 06:19:50,057 - hummingbot.market.huobi.huobi_market - NETWORK - Unexpected error while fetching account updates.
Traceback (most recent call last):
  File "huobi_market.pyx", line 489, in hummingbot.market.huobi.huobi_market.HuobiMarket._status_polling_loop
  File "/hummingbot/core/utils/async_utils.py", line 19, in safe_gather
    return await asyncio.gather(*args, **kwargs)
  File "huobi_market.pyx", line 386, in _update_order_status
  File "huobi_market.pyx", line 374, in get_order_status
  File "huobi_market.pyx", line 251, in _api_request
  File "huobi_market.pyx", line 258, in hummingbot.market.huobi.huobi_market.HuobiMarket._api_request
OSError: Error fetching data from https://api.huobi.pro/v1/order/orders/50586191544. HTTP status is 429.

in my PMM bot
2019-10-04 06:19:10,046 - hummingbot.market.huobi.huobi_market - NETWORK - Unexpected error while fetching account updates.
Traceback (most recent call last):
  File "huobi_market.pyx", line 489, in hummingbot.market.huobi.huobi_market.HuobiMarket._status_polling_loop
  File "/hummingbot/core/utils/async_utils.py", line 19, in safe_gather
    return await asyncio.gather(*args, **kwargs)
  File "huobi_market.pyx", line 386, in _update_order_status
  File "huobi_market.pyx", line 374, in get_order_status
  File "huobi_market.pyx", line 251, in _api_request
  File "huobi_market.pyx", line 258, in hummingbot.market.huobi.huobi_market.HuobiMarket._api_request
OSError: Error fetching data from https://api.huobi.pro/v1/order/orders/50584059231. HTTP status is 429.
2019-10-04 06:19:10,047 - hummingbot.market.huobi.huobi_market - WARNING - Could not fetch account updates from Huobi. Check API key and network connection.
2019-10-04 06:19:17,001 - hummingbot.strategy.pure_market_making.pure_market_making_v2 - INFO - (xzcbtc) Cancelling the limit order sell-xzcbtc-1570168311003302. [clock=2019-10-04 06:19:17+00:00]
2019-10-04 06:19:17,082 - hummingbot.market.huobi.huobi_market - ERROR - Error received from https://api.huobi.pro/v1/order/orders/50584059231/submitcancel. Response is {'status': 'error', 'err-code': 'order-orderstate-error', 'err-msg': 'Incorrect order state', 'data': None, 'order-state': 5}.
2019-10-04 06:20:04,001 - hummingbot.strategy.pure_market_making.pure_market_making_v2 - INFO - (xzcbtc) Cancelling the limit order buy-xzcbtc-1570157866001586. [clock=2019-10-04 06:20:04+00:00]
2019-10-04 06:20:04,249 - hummingbot.market.huobi.huobi_market - ERROR - Error received from https://api.huobi.pro/v1/order/orders/50569291737/submitcancel. Response is {'status': 'error', 'err-code': 'order-orderstate-error', 'err-msg': 'Incorrect order state', 'data': None, 'order-state': 5}.

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:
start the bot after import the strategy file.

Here are the steps to reproduce the issue (see attachments in the section below):
1. Set up your configuration file with these parameters: ...
2. Start the bot using this command: ...
...

**Screenshots**
// If applicable, add screenshots to help explain your problem.

**Release version**
// Include your bot's version number (Can be found at the bottom left corner of your CLI).
_docker image: coinalpha/hummingbot   development         b33825ac74cb (created 7 hrs ago)
dev-v-0.16.0_

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 
[PMM.txt](https://github.com/CoinAlpha/hummingbot/files/3689252/PMM.txt)
[XEMM.txt](https://github.com/CoinAlpha/hummingbot/files/3689253/XEMM.txt)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**
**Optional: your ETH address:** (If you want to participate in the bug bounty)

**Comment by znerol19619** [comment]
Test **XEMM & PMM** using same configuration on huobi
```
"branch": "development",
"commit": "e92cb492ea28f1a1c1f73b6605613e80d731386e",
"date": "2019-10-03T22:33:36Z",
```
Didn't get any error with regards **API** calls. Thou no orders are being made, because i don't have any balance on my acc for trading. 
![image](https://user-images.githubusercontent.com/27669951/66200645-259cd080-e6d4-11e9-8fa5-f8db41605dde.png)

Warning Message:  ``Markets are not ready. No market making trades are permitted.``

**Comment by dennisocana** [comment]
It's a known issue that our engineers are currently working on.
https://github.com/CoinAlpha/hummingbot/issues/750

It happens when an order gets partially filled and the bot can't cancel that order and then getting stuck. Since it's stuck, the bot can't create new orders to replace it.

**Comment by chowryan** [comment]
was this partial order also stuck on Huobi's website?

**Comment by rennel-tabing** [comment]
I had a conversation with the user and a work around was to recreate the strategy with the same configuration. 
The  user confirmed that orders were stuck on the actual Huobi Website.

I am adding the screenshot the user had provided 
![image](https://user-images.githubusercontent.com/53868268/67007875-56a6d900-f0b6-11e9-84c4-c203702b9373.png)

**Comment by rennel-tabing** [comment]
I am able to reproduce the other error described on the description which is 
```
Response is {'status': 'error', 'err-code': 'order-orderstate-error', 'err-msg': 'Incorrect order state', 'data': None, 'order-state': 5}.
```
On this error, I noticed that there were orders placed by hummingbot
![image](https://user-images.githubusercontent.com/53868268/67201407-3e93ca00-f3d4-11e9-9a64-adbbf3e624cd.png)

but no "Open" orders on the actual Huobi account

Here is my Hummingbot log
[IncorrrectOrderStateErrorHuobi.txt](https://github.com/CoinAlpha/hummingbot/files/3750408/IncorrrectOrderStateErrorHuobi.txt)

**Comment by dennisocana** [comment]
Fixed in 0.18.0 release, also confirmed through testing.
